kpet/run.py

```diff
@@ -32,8 +44,9 @@ def __init__(self, suite, cases):
         """
         assert isinstance(suite, data.Suite)
         assert isinstance(cases, list)
+        suite_case_names = [c.name for c in suite.cases]
         for case in cases:
-            assert case in suite.cases
+            assert case.name in suite_case_names
```

Comment:
As `cases` is not a list of `kpet.data.Case` anymore, but `kpet.run.Case`, we cannot assert like before. I don't know if the `name` is good enough or if this could fail in some circumstances.

Another solution could be making this receive a list of `kpet.data.Case`, and make the conversion inside this method (in the `self.cases = cases` below)... but I'm not sure I like it so much either, plus the `Host` constructor does receive a list of `kpet.run.Suite` objects, not a list of `kpet.data.Suite` objects. So I'm consistent with the pattern used in other places.

What do you think?

Reply:
I think let's simply drop the assertion for now. It's becoming too complicated.

I'm now thinking if we could do away with `kpet.run.Suite` altogether and just have a flat list of test case run objects right under `kpet.run.Host`, with properties from both `Suite` and `Case` aggregated (and some aggregated into `Host`). This might make it easier to write templates, randomize execution order, and other things. So let's not try too hard here.